x/marker/keeper/send_restrictions.go

```diff
@@ -146,3 +146,44 @@ func ContainsWildCard(attr string) bool {
 	segs := strings.Split(attr, ".")
 	return len(segs) > 1 && segs[0] == "*"
 }
+
+// AddToRequiredAttributes add new attributes to current list, errors if attribute already exists
+func AddToRequiredAttributes(currentAttrs []string, addAttrs []string) ([]string, error) {
+	for _, ra := range addAttrs {
+		found := false
+		for _, add := range currentAttrs {
+			if ra == add {
+				found = true
+				break
+			}
+		}
+		if found {
+			return nil, fmt.Errorf("cannot add duplicate entry to required attributes %s", ra)
+		}
+		currentAttrs = append(currentAttrs, ra)
+	}
```

Comment:
You don't need the `found` bool. You can just return the error when an existing one is found.
```suggestion
	for _, aa := range addAttrs {
		for _, ca := range currentAttrs {
			if aa == ca {
				return nil, fmt.Errorf("attribute %q is already required", aa)
			}
		}
		currentAttrs = append(currentAttrs, aa)
	}
```

If you aren't worried about `addAttrs` having duplicate entries, you can even move the append out and call it only once:
```suggestion
	for _, aa := range addAttrs {
		for _, ca := range currentAttrs {
			if aa == ca {
				return nil, fmt.Errorf("attribute %q is already required", aa)
			}
		}
	}
	currentAttrs = append(currentAttrs, addAttrs...)
```